### DISCUSSION TITLE
create extension in pure python

### DISCUSSION THREAD

**Discussion by dridk** [jupyterlab/jupyterlab#16774] [Ideas]
I'd love to be able to create extensions in pure Python.
You can already make widgets in a notebook. But I'd like to be able to take them out of the notebook. 
There's nicegui or [pyreact](https://reactpy.dev/), which may work?

**Comment by krassowski** [comment]
You may be interested in https://github.com/jtpio/ipylab

**Reply by dridk** [reply]
looks amazing !! thx
